### PR TITLE
Backend

### PR DESCRIPTION
**Pull request by furkan-kadioglu**
### Work

- Urls were defined with respect to the api was determined at [meeting](https://github.com/bounswe/bounswe2019group5/wiki/Meeting-%231-(25.09.2019))
- User model was created
- LoginView, RegisterView, GuestView were created

### Updated service

- Login 
- Register 
- Guest Login

### Comments
- I think that tests are needed. 

- backend/productionSettings.py was not pushed due to security reasons.

### CODE REVIEW

File: backend/api/views.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+class LoginView(APIView):
+    
+    def post(self,request):
+
+        if 'user_name' in request.data:
+            user = User.objects.filter(username=request.data.get('user_name')).first()
+        else:
+            user = User.objects.filter(email=request.data.get('email')).first()
+
+        if not user or not user.check_password(request.data.get('password')):
```
Comment (ymBila) [inline backend/api/views.py]: Returning token as a string of ok or not ok doesn't add much value. Same goes for message as well. I think we can remove token completely and return a message only when we need to return an error message.

File: backend/api/views.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            return Response({
+                'token':'not ok',
```
Comment (ymBila) [inline backend/api/views.py]: We should return a 4** status code indicating an error

File: backend/api/views.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+            response = {
+                'token':'ok',
+                'message':'new user is registered'
+            }
+        except:
+            response = {
+                'token':'not ok',
+                'message':'no register'
+            }
```
Comment (ymBila) [inline backend/api/views.py]: we should return a 4** status code here

File: backend/api/views.py (hunk 1 above)
Comment (furkan-kadioglu) [reply backend/api/views.py]: - Token depends on how we will handle users' session. 
- Error message should be sent when it is necessary

File: backend/api/views.py (hunk 2 above)
Comment (furkan-kadioglu) [reply backend/api/views.py]: Error will be handled 4** rest-framework status code @nevzat-ersoy-2015400090

File: backend/api/views.py (hunk 3 above)
Comment (furkan-kadioglu) [reply backend/api/views.py]: Error will be handled 4** rest-framework status code @nevzat-ersoy-2015400090

**Comment by furkan-kadioglu** [comment]
We need to fix LoginView for "test_missing_field"

**Pull request merged**